### PR TITLE
Bump spades to V4 - covert to nf-test -  add stubs

### PR DESCRIPTION
**Pull request by Joon-Klaps**
## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [ ] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

**Review by nvnieuwk — APPROVED** [review]
LGTM

File: modules/nf-core/spades/tests/main.nf.test [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+            process {
+                """
+                input[0] = [ [ id:'test', single_end:false ], // meta map
+                            [ file(params.modules_testdata_base_path + "genomics/sarscov2/illumina/fastq/test_1.fastq.gz", checkIfExists: true),
+                                file(params.modules_testdata_base_path + "genomics/sarscov2/illumina/fastq/test_2.fastq.gz", checkIfExists: true) ],
+                            [],
+                            [ file(params.modules_testdata_base_path + "genomics/sarscov2/nanopore/fastq/test.fastq.gz", checkIfExists: true) ]
+                        ]
+                input [1] = file(params.modules_testdata_base_path + "delete_me/spades/spades_input_yml.yml", checkIfExists: true)
+                input [2] = []
```
Comment (itrujnara) [inline modules/nf-core/spades/tests/main.nf.test:152]: I don't see any test that would use input[2] (`hmm`). Is it possible to add one?

**Review by itrujnara — COMMENTED** [review]
Nice one! I see no problems in the module itself. The tests are really comprehensive, but I think there is one missing.

File: modules/nf-core/spades/tests/main.nf.test (hunk 1 above)
Comment (Joon-Klaps) [reply modules/nf-core/spades/tests/main.nf.test:152]: You are right, there should be one. I added hmm test.

**Review by itrujnara — APPROVED** [review]
Thanks, looks good now

**Pull request merged**